streams/writable-streams/general.js

```diff
@@ -119,48 +119,48 @@ promise_test(() => {
       resolvers[methodName] = resolve;
     });
   }
-
+  var thisObject = null;
   // Calls to Sink methods after the first are implicitly ignored. Only the first value that is passed to the resolver
   // is used.
   class Sink {
     start() {
-      // Called twice
+      assert_equals(thisObject, this, 'start should be called as a method');
```

Comment:
asserts don't work properly in sink methods because the exceptions they throw are caught by the WritableStream. This is a serious trap: we used to have lots of tests that looked fine but weren't actually testing anything.

I mostly fixed these by moving the asserts out of the methods, but another way that works is to replace `start() { assert_something(); }` with `start: t.step_func(() => { assert_something(); })`.